jsregexp.c

```diff
@@ -262,10 +269,172 @@ static int regexp_tostring(lua_State *lstate)
 }
 
 
+// automatic conversion to the global match string
+static int match_tostring(lua_State *lstate)
+{
+  //luaL_getmetatable(lstate, JSREGEXP_MATCH);
+  //if (!lua_getmetatable(lstate, 1) || !lua_equal(lstate, -1, -2)) {
+  //  luaL_argerror(lstate, 1, "match object expected");
+  //}
+  lua_rawgeti(lstate, 1, 0);
+  return 1;
+}
+
+
+// repeatedly running regexp:match(str) is not a good idea because we would
+// convert the string (at least from last_ind) to utf16 every time (if it is
+// needed)
+static int regexp_exec(lua_State *lstate)
+{
+  uint8_t *capture[CAPTURE_COUNT_MAX * 2];
+
+  size_t input_len;
+  struct regexp *r = luaL_checkudata(lstate, 1, JSREGEXP_MT);
+  const char *input = luaL_checklstring(lstate, 2, &input_len);
+
+  // check what happens in JS
+  if (r->last_index >= input_len) {
```

Comment:
based on quickjs if `(r->last_index > input_len)` `r->last_index` should be set to 0. I feel it should be `>=`, but that is what the code shows.

Reply:
Ah, it actually throws an exception in that case. `>` makes sense in that if the previous match ended at the end of the input, the next call to `exec` should return `null`.